### PR TITLE
Bank refund functionality v1

### PR DESCRIPTION
**Pull request by sauravmaiti22**
Implemented bank refund functionality for bank payment attempts. Following cases tested:
1. Single bank refund.
2. Partial bank refund.
3. Payout delete -> Refund cancelled
4. Payout create -> Refund pending approval
5. Payout approve -> Refunded

**To test this in dev use nofrixion bank to make payment using a different merchant. This way the amount will get settled. Difficult to test with other banks.**

Version 2 - Need to work on approval flow after changes in bff. Will create an item for that.

### CODE REVIEW

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/32.

File: packages/components/src/components/ui/BankRefundModal/BankRefundModal.tsx [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+      onDismiss()
+    }
+  }
+
+  return (
+    <>
+      {bankPaymentAttempt && paymentRequest && (
+        <Sheet open={!!bankPaymentAttempt} onOpenChange={handleOnOpenChange}>
+          <SheetContent className="w-full lg:w-[37.5rem]">
+            <div className="bg-white max-h-screen overflow-auto">
```
Comment (donalnofrixion) [inline packages/components/src/components/ui/BankRefundModal/BankRefundModal.tsx:102]: Are all these divs needed? They seem to share css. Can they be merged into one div?

File: packages/components/src/components/ui/molecules/Select/SelectAccount/SelectAccount.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+              )}
+            </div>
+          </SelectValue>
+        </SelectTrigger>
+      }
+      <SelectContent className="w-full md:w-[375px] z-[200]">
+        {accounts.map((account) => (
+          <SelectItem key={account.id} value={account.id} isText={false}>
+            <div className="w-full flex justify-between">
+              <>
```
Comment (donalnofrixion) [inline packages/components/src/components/ui/molecules/Select/SelectAccount/SelectAccount.tsx:63]: This fragment isn't needed

**Review by donalnofrixion — CHANGES_REQUESTED** [review]

File: packages/components/src/components/ui/BankRefundModal/BankRefundModal.tsx (hunk 1 above)
Comment (sauravmaiti22) [reply packages/components/src/components/ui/BankRefundModal/BankRefundModal.tsx]: Yes they can be merged. I have made the changes to do that.

File: packages/components/src/components/ui/molecules/Select/SelectAccount/SelectAccount.tsx (hunk 2 above)
Comment (sauravmaiti22) [reply packages/components/src/components/ui/molecules/Select/SelectAccount/SelectAccount.tsx]: Removed it.

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/32.

**Review by pabloi-nf — CHANGES_REQUESTED** [review]
Apply fixes on the image
image

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/32.

**Comment by pabloi-nf** [comment]
Thanks @sauravmaiti22 it all looks good. 
There is only one last thing I'm noticing, that is an extra space under the "to account" name
image

If we remove that 0.5rem margin we are good.
Thanks

**Review by pabloi-nf — CHANGES_REQUESTED** [review]
Thanks @sauravmaiti22 it all looks good.
There is only one last thing I'm noticing, that is an extra space under the "to account" name
image

If we remove that 0.5rem margin we are good.
Thanks

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/32.

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/32.

**Review by pabloi-nf — APPROVED** [review]

**Comment by sipsorcery** [comment]
Latest changes for this PR can be tested at https://business-dev.nofrixion.com/32.

**Review by arifmatin — APPROVED** [review]

**Review by donalnofrixion — APPROVED** [review]

**Pull request merged**